compiler/semexprs.nim

```diff
@@ -1674,6 +1674,9 @@ proc semAsgn(c: PContext, n: PNode; mode=asgnNormal): PNode =
         var rhsTyp = rhs.typ
         if rhsTyp.kind in tyUserTypeClasses and rhsTyp.isResolvedUserTypeClass:
           rhsTyp = rhsTyp.lastSon
+        if isClosureIterator(c.p.owner) and
+            cmpTypes(c, c.p.owner.typ.sons[0], rhsTyp) notin {isGeneric, isEqual}:
```

Comment:
Why is this ``cmpTypes`` call required here?

Reply:
At one point it prevented a segfault, but no longer necessary if the CIs agree.